sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_queue.cpp

```diff
@@ -1389,6 +1389,23 @@ ur_result_t ur_queue_handle_t_::synchronize() {
     // event.
     if (isInOrderQueue() && !LastCommandEvent->IsDiscarded) {
       ZE2UR_CALL(zeHostSynchronize, (LastCommandEvent->ZeEvent));
+
+      // clean up all events known to have been completed as well,
+      // so they can be reused later
+      for (auto &QueueMap : {ComputeQueueGroupsByTID, CopyQueueGroupsByTID}) {
+        for (auto &QueueGroup : QueueMap) {
+          if (Device->ImmCommandListUsed) {
+            for (auto ImmCmdList : QueueGroup.second.ImmCmdLists) {
+              if (ImmCmdList == this->CommandListMap.end())
+                continue;
+              // Cleanup all events from the synced command list.
+              auto EventListToCleanup = std::move(ImmCmdList->second.EventList);
```

Comment:
I wonder if we could avoid copying the list of events (it may be large with immediate submissions)

Reply:
what about just doing it in the same vector

```cpp
diff --git a/sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_queue.cpp b/sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_queue.cpp
index 2bab64598297..479810b88878 100644
--- a/sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_queue.cpp
+++ b/sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero_queue.cpp
@@ -1399,9 +1399,8 @@ ur_result_t ur_queue_handle_t_::synchronize() {
               if (ImmCmdList == this->CommandListMap.end())
                 continue;
               // Cleanup all events from the synced command list.
-              auto EventListToCleanup = std::move(ImmCmdList->second.EventList);
+              CleanupEventListFromResetCmdList(ImmCmdList->second.EventList, true);
               ImmCmdList->second.EventList.clear();
-              CleanupEventListFromResetCmdList(EventListToCleanup, true);
             }
           }
         }
```

Reply:
Let's try this. I don't recall why a copy was used in the first place.

Reply:
@smaslov-intel : updated as suggested.